metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1065,6 +1065,11 @@ public void handleCommit(BatchReader<ApiMessageAndVersion> reader) {
                         int epoch = batch.epoch();
                         List<ApiMessageAndVersion> messages = batch.records();
 
+                        if (messages.isEmpty()) {
+                            log.debug("Skipping handling commit for batch with no data records with offset {} and epoch {}.", offset, epoch);
+                            continue;
```

Comment:
`continue` is kind of a messy construct. Can we do something like this instead?
```
if (messages.isEmpty()) {
    log.debug("Skipping handling commit for batch with no data records with offset {} and epoch {}.", offset, epoch);
} else if (isActive) {
...
```

Reply:
Also, we should update the metrics in this case (should be easy to do by adding a new method to OffsetsControlManager to do just that and nothing else)